src/keygen/KeygenOptions.hs

```diff
@@ -0,0 +1,93 @@
+{-# LANGUAGE ApplicativeDo #-}
+
+module KeygenOptions
+       ( KeygenOptions (..)
+       , AvvmStakeOptions (..)
+       , TestStakeOptions (..)
+       , optsInfo
+       ) where
+
+import           Options.Applicative (Parser, ParserInfo, auto, fullDesc, help, helper,
+                                      info, long, metavar, option, option, progDesc,
+                                      short, strOption, switch, value)
+import           Universum
+
+data KeygenOptions = KO
+    { koGenesisFile :: FilePath
+    , koTestStake   :: Maybe TestStakeOptions
+    , koAvvmStake   :: Maybe AvvmStakeOptions
+    }
+
+data TestStakeOptions = TSO
+    { tsoPattern      :: FilePath
+    , tsoPoors        :: Word
+    , tsoRichmen      :: Word
+    , tsoRichmenShare :: Double
+    , tsoTotalStake   :: Word64
+    }
+
+data AvvmStakeOptions = ASO
+    { asoJsonPath      :: FilePath
+    , asoIsRandcerts   :: Bool
+    , asoHolderKeyfile :: Maybe FilePath
+    }
+
+optsParser :: Parser KeygenOptions
+optsParser = do
+    koGenesisFile <- strOption $
+        long    "genesis-file" <>
+        metavar "FILE" <>
+        value   "genesis.bin" <>
+        help    "File to dump binary shared genesis data"
+    koTestStake <- optional testStakeParser
+    koAvvmStake <- optional avvmStakeParser
+    pure KO{..}
+
+testStakeParser :: Parser TestStakeOptions
+testStakeParser = do
+    tsoPattern <- strOption $
+        long    "file-pattern" <>
+        short   'f' <>
+        metavar "PATTERN" <>
+        help    "Filename pattern for generated keyfiles \
+                \(`{}` is a place for number)"
+    tsoPoors <- option auto $
+        long    "testnet-keys" <>
+        short   'n' <>
+        metavar "INT" <>
+        help    "Number of testnet stakeholders to generate"
+    tsoRichmen <- option auto $
+        long    "richmen" <>
+        short   'm' <>
+        metavar "INT" <>
+        help    "Number of rich stakeholders to generate"
+    tsoRichmenShare <- option auto $
+        long    "richmen-share" <>
+        metavar "FLOAT" <>
+        value   0.7 <>
```

Comment:
Can you please remove it if not necessary? It's better to make tricky args obligatory than placing default values that can't be understood quickly.